src/basic/memory-util.c

```diff
@@ -39,3 +39,17 @@ bool memeqbyte(uint8_t byte, const void *data, size_t length) {
         /* Now we know first 16 bytes match, memcmp() with self.  */
         return memcmp(data, p + 16, length) == 0;
 }
+
+void *make_swapped(const void *data, size_t size) {
```

Comment:
so what does this do? make a copy of the data and reverse the order of the bytes in the copy?

maybe call this "memreverse()"? or so?

Reply:
IMHO the prefix `make_` clarifies that the function allocates a new buffer instead of swapping bytes in-place. `make_reversed` also sounds good. `memreverse` suits more for in-place modification.

> so what does this do?

Add a comment about it?

Reply:
"swapped" sounds like byteswapped to me, i.e. LE/BE conversion...

maybe call thus memdup_reverse() or so? it is very similar to the existing memdup() call we have after all

Reply:
and comment would be good too, of course. comments are always good.

Reply:
The comment in the header already.